homeassistant/components/samsungtv/config_flow.py

```diff
@@ -78,15 +77,77 @@ def _get_entry(self):
             data=data,
         )
 
+    async def _async_set_device_unique_id(self, raise_on_progress=True):
+        """Set device unique_id."""
+
+        if self._id:
+            await self.async_set_unique_id(
+                self._id, raise_on_progress=raise_on_progress
+            )
+            self._abort_if_unique_id_configured()
+
+        if not self._device_info:
+            await self._async_get_and_check_device_info()
+
+        if uuid := self._device_info.get("device", {}).get(ATTR_UPNP_UDN.lower()):
+            self._id = uuid
+
+        if not self._id:
+            LOGGER.debug(
+                "Property " "%s" " is missing for host %s",
+                ATTR_UPNP_UDN.lower(),
+                self._host,
+            )
+            return self.async_abort(reason=RESULT_ID_MISSING)
+
+        if self._id.startswith("uuid:"):
+            self._id = self._id[5:]
+
+        await self.async_set_unique_id(self._id, raise_on_progress=raise_on_progress)
+        self._abort_if_unique_id_configured()
+
     def _try_connect(self):
         """Try to connect and check auth."""
+        if self._bridge:
+            return
+
         for method in SUPPORTED_METHODS:
             self._bridge = SamsungTVBridge.get_bridge(method, self._host)
             result = self._bridge.try_connect()
+            if result == RESULT_SUCCESS:
+                return
             if result != RESULT_CANNOT_CONNECT:
-                return result
+                raise data_entry_flow.AbortFlow(result)
         LOGGER.debug("No working config found")
-        return RESULT_CANNOT_CONNECT
+        raise data_entry_flow.AbortFlow(RESULT_CANNOT_CONNECT)
+
+    async def _async_get_and_check_device_info(self):
+        """Try to get the device info."""
+        if self._bridge:
+            self._device_info = await self.hass.async_add_executor_job(
+                self._bridge.device_info
+            )
+        else:
+            for port in WEBSOCKET_PORTS:
+                self._device_info = await self.hass.async_add_executor_job(
+                    SamsungTVBridge.get_bridge(
+                        METHOD_WEBSOCKET, self._host, port
+                    ).device_info
+                )
+                if self._device_info:
+                    break
+
+            if not self._device_info:
+                raise data_entry_flow.AbortFlow(RESULT_NOT_SUPPORTED)
```

Comment:
We should do this all in one executor job, and lower the timeouts to 5s per method

Reply:
5s for the connection are indeed more than enough, but you will receive an authorization prompt on the TV and the user may take a bit to respond.

Reply:
Aren't these running without avoid_open?

Reply:
Fixed by https://github.com/bdraco/home-assistant/commit/9073b4a8f18ad449d52dd03689f85ba0ddb16272